R/symptome_compartment_class.R

```diff
@@ -0,0 +1,226 @@
+              
+#' @describeIn model_symptome set initial population sizes (in fraction) and parameters
+#'
+#' @param object An object of class model_symptome
+#' @param S Numeric, initial population of the susceptible
+#' @param E Numeric, initial population of the exposed
+#' @param I Numeric, initial population of the infected with mild/no symptom
+#' @param HA Numeric, initial population of the infected with severe symptoms
+#' that need further hospitalization
+#' @param CM Numeric, initial cumulative mortality
+#' @param R Numeric, initial population of the recovered
+#' @param lam Numeric, rate at which an infected individual exposes susceptible
+#' @param gamma Numeric, rate at which exposed individuals become infected
+#' @param omega Numeric, rate at which recovered individuals become susceptible
+#' @param ihr Numeric, fraction of infected individuals develop severe symptoms
+#' @param nu Numeric vector, rate at which each infected group recover
+#' @param pdeath Numeric vector, rate of disease-caused mortality of each infected group
+#' @return An object of class model_symptome with initial population and parameters
+#'
+setGeneric(name = "set_init",
+           def = function(object, S=NA_real_, E=NA_real_, I=NA_real_,
+                          HA=NA_real_, R=NA_real_, CM=NA_real_,
+                          lam=NA_real_, gamma=NA_real_, omega=NA_real_,
+                          ihr=NA_real_, nu=list(), pdeath=list()) standardGeneric("set_init"))
+setMethod("set_init", signature(object = "model_symptome"),
+          function(object, S, E, I, HA, R, CM, lam, gamma, omega, ihr, nu, pdeath) {
+            param_list <- list(lam, gamma, omega, ihr, nu, pdeath)
+            init_pop_list <- list(S, E, I, HA, R, CM)
+            names(param_list) <- names(object@parameters)
+            names(init_pop_list) <- names(object@initial_population)
+            object@initial_population <- init_pop_list
+            object@parameters <- param_list
+
+            # check if initial settings are valid
+            check <- check_init(object)
+            if (check == TRUE) object
+            else stop(paste(check, ", please check and rerun set_init.\n"))
+          })
+
+#' @describeIn model_symptome to check validity of the settings of the parameters and initial conditions
+#'
+#' @param object An object of class model_symptome
+#' @return TRUE if passing all checks, or an error message if any check fails
+#'
+setGeneric(name = "check_init", def = function(object) standardGeneric("check_init"))
+setMethod("check_init", signature(object = "model_symptome"),
+          function(object) {
+            errors <- character()
+            # check whether all required parameters are set
+            is_na_params <- is.na(object@parameters)
+            if (sum(is_na_params) != 0) {
+              msg <- paste("Missing parameters:",
+                           paste(names(object@parameters)[is_na_params], collapse = ", ")
+              )
```

Comment:
I think we are just using the stop() function at the moment to throw errors. Is there a good reason to use one over the other?

Reply:
I think this is ok because these are now collected into a stop later on.